heat/core/_operations.py

```diff
@@ -43,6 +44,8 @@ def __binary_op(
         The second operand involved in the operation,
     out: DNDarray, optional
         Output buffer in which the result is placed
+    where: DNDarray, optional
+        Condition of interest, where true yield the result of the operation else yield original value in out (uninitialized when out=None)
```

Comment:
We can use numpy's docs for `where`, I think they are a bit clearer. But we must expand on them a bit, e.g. is `where` supposed/expected to be distributed, and how.